lib/AnalysisStructured/PtrAnalysis.cpp

```diff
@@ -225,13 +225,33 @@ LogicalResult PtrState::addState(const PtrState &lhsState,
       strides.push_back(builder.getIndexAttr(1));
       // New offset is offset * stride.
       auto newLhsOffset = lhsState.offsets[i];
-      if (!hasConstZero(lhsState.strides[i])) {
+      if (lhsState.hasModulo()) {
+        auto mod = expandOFRIndex(lhsState.shape[i], lhsState.offsets[i], loc,
+                                  builder);
+        // Apply modulo to the offset.
+        newLhsOffset = remOFRs(lhsState.offsets[i], mod, loc, builder);
+      }
+      // When the dimension is structured, mul the offset by the stride to match
+      // the stride 1 for non-structured dimensions.
+      // If the dimension is not structured, the offset is already
```

Comment:
this section is getting very hard to understand. could you please add a sample triton code here to make things more readable